assets/js/components/user-input/UserInputSelectOptions.js

```diff
@@ -178,17 +186,52 @@ export default function UserInputSelectOptions( {
 		}
 
 		return (
-			<div
-				key={ optionSlug }
-				className="googlesitekit-user-input__select-option"
-			>
-				<ListComponent { ...props }>
-					{ options[ optionSlug ] }
-				</ListComponent>
-			</div>
+			'sell_products_or_service' !== optionSlug && (
+				<div
+					key={ optionSlug }
+					className="googlesitekit-user-input__select-option"
+				>
+					<ListComponent { ...props }>
+						{ options[ optionSlug ] }
+					</ListComponent>
+				</div>
+			)
 		);
 	} );
 
+	const currentlyEditingSlug = useSelect( ( select ) =>
+		select( CORE_UI ).getValue( USER_INPUT_CURRENTLY_EDITING_KEY )
+	);
+
+	const previousValues = usePrevious( values );
+
+	useEffect( () => {
+		if (
+			currentlyEditingSlug === USER_INPUT_QUESTIONS_PURPOSE &&
+			values.includes( 'sell_products_or_service' ) &&
+			'sell_products' in options
```

Comment:
This extra check seems to be redundant, if we are currently on edit screen within `USER_INPUT_QUESTIONS_PURPOSE` and legacy `sell_products_or_service` option is saved, it should be fairly enough to determine we need to change answer, and `sell_products` as such - a core answer is allways going to be present, as it is a hardcoded value in a constant

Reply:
Thanks @zutigrm resolved in a recent commit.